api/serializers.py

```diff
@@ -1,20 +1,25 @@
-from rest_framework import serializers
-from api.models import TaskDef, Task, STATUS_CHOICES, PRIORITY_CHOICES
+from rest_framework import serializers, exceptions
+from django.db.utils import IntegrityError
+from api.models import TaskDef, Task, PRIORITY_CHOICES
+
+class UniqueTaskConflict(exceptions.APIException):
+    status_code = 409
+    default_detail = 'Task `unique` field conflict'
 
 class TaskDefSerializer(serializers.Serializer):
     name = serializers.CharField(required=True, allow_blank=False, max_length=255)
-    priority_levels = serializers.ListField(
-        child=serializers.ChoiceField(choices=PRIORITY_CHOICES)
-    )
-    title = serializers.CharField(required=False, allow_blank=False, max_length=255)
-    description = serializers.CharField(required=False, allow_blank=False, max_length=2048)
+    title = serializers.CharField(required=False, allow_null=True, allow_blank=False, max_length=255)
+    description = serializers.CharField(required=False, allow_null=True, allow_blank=False, max_length=2048)
     default_timeout = serializers.IntegerField(required=False)
     max_attempts = serializers.IntegerField(required=False)
     created_at = serializers.DateTimeField(read_only=True, format='iso-8601')
     updated_at = serializers.DateTimeField(read_only=True, format='iso-8601')
 
     def create(self, validated_data):
-        return TaskDef.objects.create(**validated_data)
+        try:
+            return TaskDef.objects.create(**validated_data)
+        except IntegrityError:
+            raise exceptions.ValidationError({'name': '"' + validated_data['name'] + '" already taken.'})
```

Comment:
@dhimmel ValidationError requires a dict or list of dicts. The idea being that each key is the field with the error, then a pretty formatter could interpret it on the other end.